src/hawkbit-client.c

```diff
@@ -245,17 +245,24 @@ static gboolean get_binary(const gchar* download_url, const gchar* file, gint64
 }
 
 /**
- * @brief Curl callback used for writting rest response to buffer.
+ * @brief Curl callback writing REST response to RestPayload*->payload buffer.
+ *
+ * @see   https://curl.haxx.se/libcurl/c/CURLOPT_WRITEFUNCTION.html
  */
-static size_t curl_write_cb(void *content, size_t size, size_t nmemb, void *data)
+static size_t curl_write_cb(const void *content, size_t size, size_t nmemb, void *data)
 {
-        RestPayload *p = (RestPayload *) data;
+        RestPayload *p = NULL;
         size_t real_size = size * nmemb;
 
+        g_return_val_if_fail(content, 0);
+        g_return_val_if_fail(data, 0);
+
+        p = (RestPayload *) data;
+
         p->payload = (gchar *) g_realloc(p->payload, p->size + real_size + 1);
-        if (p->payload == NULL) {
-                g_debug("Failed to expand buffer");
-                return -1;
+        if (!p->payload) {
+                g_error("Failed to expand buffer");
+                return 0;
```

Comment:
return after g_error will likely never be called but abort the program.

Reply:
I guess it is okay that a failing `g_realloc()` aborts. We can't recover from that. The return statement is there for the sake of understandability.

Reply:
You do not neet to check for the return actually as both the function description as well as the implementation do not state to signalize errors with returning NULL.
Instead, the function will fail itself with g_error() when allocatio fails. Refer: https://gitlab.gnome.org/GNOME/glib/-/blob/master/glib/gmem.c#L176

Reply:
Right, dropped the `if` altogether